### DISCUSSION TITLE
Unable to install Foundry

### DISCUSSION THREAD

**Discussion by S0mt0** [Cyfrin/foundry-full-course-cu#2700] [Q&A] [answered]
Hi,
I have tried installing Foundry on my windows PC using the command suggested on the resources page and also on the Foundry Book documentation:
```
curl -L https://foundry.paradigm.xyz | bash
```

However, I keep getting the SSL timeout report/error as in the following:

```
sewkito@Somto:~$ curl -L https://foundry.paradigm.xyz | bash
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100   167  100   167    0     0    235      0 --:--:-- --:--:-- --:--:--   235
  0     0    0     0    0     0      0      0 --:--:--  0:05:01 --:--:--     0
curl: (28) SSL connection timeout
```

This is not an internet connection issue because every other page or app using the internet works fine.
Please, what do I do?

Can somebody help me?
<img width="657" alt="ubuntu terminal - foundry installation" src="https://github.com/user-attachments/assets/a75551a4-600b-4739-8f1a-95522a222033">

**Comment by EngrPips** [comment]
Hello @S0mt0, This seem like an internet connection problem, Please retry when your internet connection is stable.

**Reply by S0mt0** [reply]
@EngrPips Thank you for reaching out. 
But my internet is pretty stable; I've even had to use the same internet connection to install different versions of ubuntu, hoping it'd fix the problem but the error keeps persisting😔

![image](https://github.com/user-attachments/assets/00ea2473-53a5-4832-9994-dbb105e4cf16)

**Reply by EngrPips** [reply]
I didn't run into this issue and I am unsure what is wrong but please check the official foundry repo https://github.com/foundry-rs/foundry if a similar issue is reported.

**Reply by S0mt0** [reply, marked as answer]
@EngrPips Thank you for sharing with me the foundry github link where I went through their discussions and found a few persons who had encountered similar issues. So as suggested by one of the individuals, I just had to change my network provider and also turned of my public firewall. However, I would say, I'm honestly not sure which the fix was--changing internet network provider or temporarily turning off my firewall😂(though I'd favor changing internet provider more cos I think I had already tried the firewall approach earlier)--but I'm glad it worked either ways.

Here's a link to the issue where I got the suggestion from: https://github.com/foundry-rs/foundry/discussions/5775#discussioncomment-8122875.

![image](https://github.com/user-attachments/assets/e1966643-0291-49f7-af94-da189894c51d)

**Reply by EngrPips** [reply]
Awesome, I am glad you took the time to check the repo and find the solution. Many people are too lazy even to help themself.

**Reply by S0mt0** [reply]
Thank you 😌

**Reply by EngrPips** [reply]
Anytime.
